pycheribuild/config/target_info.py

```diff
@@ -581,33 +577,33 @@ def target_triple(self):
         assert self.target.is_riscv(include_purecap=True)
         return "riscv64-unknown-rtems" + str(self.RTEMS_VERSION)
 
-    @property
-    def sdk_root_dir(self):
-        return self.config.cheri_sdk_dir
-
     @property
     def sysroot_dir(self):
-        return Path(self.sdk_root_dir, self.target_triple)
+        # Install to target triple as RTEMS' LLVM/Clang Driver expects
+        return self.sdk_root_dir / ("sysroot-" + self.target.generic_suffix) / self.target_triple
```

Comment:
We should be passing --sysroot to all compile jobs so the `if (!D.SysRoot.empty()) {` branch will be taken.
Relying on the implicit sysroot detection seems fragile to me: what happens if we build rtems for non-CHERI RISCV? It might pick the wrong one since the triple is always the same.

Reply:
That's really the same behavoir with GCC. You have to make sure you build/use the correct default libraries in _sdk_root/riscv64_unknown-rtems5_ (or until we have a multi-lib support). I have had this discussion with other RTEMS developers and we agreed to do that.

RTEMS build system itself (i.e., not from cheribuild) can still provide its own --sysroot (if it wants), in which case the `if (!D.SysRoot.empty())` will be taken if there are any ambiguity concerns. In fact the autotools build system used to do so, but we chose not to go this way.

Reply:
So you can't use the compiler in /usr/bin unless you have write access to /usr/riscv64_unknown-rtems5? I don't like that design choice. Why can't we just pass --sysroot to the build?

Reply:
That's the discussion we had about sysroot [1]. We tried to make things look like RTEMS' GCC as possible (in which case _/usr/riscv64_unknown-rtems5_ would have been installed anyway when you properly build the RTEMS Clang/LLVM toolchain from RSB [2]). Furthermore, RTEMS has its own framework to build its toolchain in given path/directories that it has access to (to avoid using inconsistent native tools like in /usr/). Relying on clang in $PATH shouldn't really be the normal case, rather, it will search for its tools from --rtems-tools flags.

[1] https://lists.rtems.org/pipermail/devel/2019-November/056129.html
[2] https://git.rtems.org/rtems-source-builder/tree/rtems/config/tools/rtems-llvm-8.0.1.cfg

Reply:
I don't see the problem with multilib. --sysroot should just be equivalent to pretending that --sysroot is equivalent to "/" so if the driver expects a triple it can just look in $SYSROOT/$triple.

And while there isn't native multilib support in the driver you can just pass a different --sysroot flag for a sysroot with a different target.

I see that RTEMS likes to build the whole toolchain so they can do whatever but
I would really like to have all sysroots for all supported operating systems in a `sysroot-foo` directory in the CHERI SDK.